docs/specs/delegation.md

```diff
@@ -0,0 +1,110 @@
+
+### Space Admin Space
+
+```
+DELEGATIONS = UCAN[];
+// Mapping of charm name to charm identity
+ITEMS = Map<string, DidKey>
+```
+
+## Capabilities
+
+Access and permissions are handled via an authorization token like [UCAN](https://github.com/ucan-wg/spec) or another alternative. These tokens represent capabilities delegated to identities for a given subject.
+
+* `SPACE:CREATE`: A namespace-based capability indicating permission to create a new space. This is the only delegation stored in a namespace admin record. 
+* `SPACE:READ`: Permission to read a space. 
+* `SPACE:WRITE`: Permission to write data to a space. A superset of `SPACE:READ`.
+* `SPACE:OWNER`: Permission to modify authorization in a space. A superset of `SPACE:WRITE`.
+
+> [!NOTE]
+> Revocations and rotations are currently out of scope, but could be handled with delegation.
+
+### The `ANYONE` User
+
+We use a well-known key to assign delegation to all users.
+
+### Example
+
+Alice (`did:key:alice`) creates a new space on `provider.com` (with identity `did:key:provider`) with the name `alice-space`. No namespace was provided, so the *provider namespace* is used. The space name with identity `did:key:provider` (possibly a derivation of identity) is referenced to ensure Alice has permission to create a new space, and that `alice-space` is an available space name.
+
+`provider.com` was configured to allow all users to create a new space by delegating the `SPACE:CREATE` capability to the `ANYONE` identity, stored in the `did:key:provider` space's delegations. Additionally, there are no current spaces with the name `alice-space`. The provider creates the space on Alice's behalf, generating a new key for the space (`did:key:alice-space`). This key is stored by the provider, immediately delegating `SPACE:OWNER` capabilities to `did:key:alice`.
+
+Now, `provider.com` is hosting `alice-space` in the provider namespace. No one else yet has any access to data stored in this space. Alice wants to invite Bob (`did:key:bob`) to this space, and signs a new delegation with `did:key:alice` for `did:key:bob`, granting Bob `SPACE:WRITE` permissions. Bob can now read and write data to the space, but cannot invite Eve, lacking the `SPACE:OWNER` capability.
+
+Similarly, in a non-provider namespace (e.g. `@alice.fab.com`, or `@did:key:alice`), the namespace is created, immediately assigning `SPACE:CREATE` capabilities to `did:key:alice`, or whatever key `@alice.fab.com` resolves to. No other identities may create spaces within Alice's namespace, unless they were to e.g. delegate `SPACE:CREATE` capabilities to another user.
+
+## Plan
+
+Currently, `Session`s are created in the workspace, and the abstraction between public/private spaces needs updated. There are a few areas currently that will need to use a new Session compatible with this document:
```

Comment:
I think we can start with today's derived-from-name spaces for now and focus on adding ACLs.

Effectively ~ partitions the namespaces per user, but we could just choose to not use that.

In a way it's the most simple implementation of provider-level lookup and we can just grow it from there.